### DISCUSSION TITLE
error handling istio-validator-istio-system, retrying (retry count: 33): webhook is not ready

### DISCUSSION THREAD

**Discussion by yuvrajsinh-chauhan1** [istio/istio#48925] [Q&A]
Hi,

I have installed Kubernetes on EC2 and below is the version details-
Client Version: v1.28.2
Kustomize Version: v5.0.4-0.20230601165947-6ce0bf390ce3
Server Version: v1.28.2
Istion Version - 1.20.2 (installed using demo profile)

I have installed Istio using istioctl and installation was successful and all pods are in running state, but when i checked logs from istiod pod it encountered with error - 
---------------------------------------------------------------------------------------------------------------
```
2024-01-23T14:30:49.890934Z     info    ads     EDS: PUSH request for node:istio-ingressgateway-778f947c84-wlttv.istio-system resources:31 size:5.9kB empty:0 cached:31/31
2024-01-23T14:30:49.891041Z     info    ads     LDS: PUSH request for node:istio-ingressgateway-778f947c84-wlttv.istio-system resources:0 size:0B
2024-01-23T14:31:33.413410Z     info    validationController    Not ready to switch validation to fail-closed: dummy invalid config not rejected
2024-01-23T14:31:33.413474Z     info    validationController    validatingwebhookconfiguration istio-validator-istio-system (failurePolicy=Ignore, resourceVersion=14048700) is up-to-date. No change required.
2024-01-23T14:31:33.413495Z     error   controllers     error handling istio-validator-istio-system, retrying (retry count: 128): webhook is not ready, retrycontroller=validation
2024-01-23T14:31:40.743728Z     info    validationController    Not ready to switch validation to fail-closed: dummy invalid config not rejected
2024-01-23T14:31:40.743788Z     info    validationController    validatingwebhookconfiguration istiod-default-validator (failurePolicy=Ignore, resourceVersion=14048817) is up-to-date. No change required.
2024-01-23T14:31:40.743810Z     error   controllers     error handling istiod-default-validator, retrying (retry count: 128): webhook is not ready, retry    controller=validation
```
--------------------------------------------------------------------------------------------------
```
kubectl get mutatingwebhookconfigurations,validatingwebhookconfigurations -n istio-system
NAME                                                                                   WEBHOOKS   AGE
mutatingwebhookconfiguration.admissionregistration.k8s.io/istio-revision-tag-default   4          123m
mutatingwebhookconfiguration.admissionregistration.k8s.io/istio-sidecar-injector       4          123m

NAME                                                                                       WEBHOOKS   AGE
validatingwebhookconfiguration.admissionregistration.k8s.io/istio-validator-istio-system   1          123m
validatingwebhookconfiguration.admissionregistration.k8s.io/istiod-default-validator       1          123m
```
-----------------------------------------------------------------------------------------------------------------------
Error while creating a new pod in default namespace - 
```
kubectl apply -f nginx.yaml
Error from server (InternalError): error when creating "nginx.yaml": Internal error occurred: failed calling webhook "namespace.sidecar-injector.istio.io": failed to call webhook: Post "https://istiod.istio-system.svc:443/inject?timeout=10s": Forbidden
```
------------------------------------------------------------------------------------------------------------------------
Kube API Error Logs -
```
W0123 09:48:47.972657       1 dispatcher.go:225] Failed calling webhook, failing closed namespace.sidecar-injector.istio.io: failed calling webhook "namespace.sidecar-injector.istio.io": failed to call webhook: Post "https://istiod.istio-system.svc:443/inject?timeout=10s": Forbidden
I0123 09:48:47.993590       1 alloc.go:330] "allocated clusterIPs" service="default/nginx-service" clusterIPs={"IPv4":"10.100.197.47"}
W0123 09:49:10.796965       1 dispatcher.go:205] Failed calling webhook, failing open rev.validation.istio.io: failed calling webhook "rev.validation.istio.io": failed to call webhook: Post "https://istiod.istio-system.svc:443/validate?timeout=10s": Forbidden
E0123 09:49:10.797076       1 dispatcher.go:213] failed calling webhook "rev.validation.istio.io": failed to call webhook: Post "https://istiod.istio-system.svc:443/validate?timeout=10s": Forbidden
W0123 09:49:17.840757       1 dispatcher.go:205] Failed calling webhook, failing open rev.validation.istio.io: failed calling webhook "rev.validation.istio.io": failed to call webhook: Post "https://istiod.istio-system.svc:443/validate?timeout=10s": Forbidden
E0123 09:49:17.840871       1 dispatcher.go:213] failed calling webhook "rev.validation.istio.io": failed to call webhook: Post "https://istiod.istio-system.svc:443/validate?timeout=10s": Forbidden
```
-----------------------------------------------------------------------------------------------------------------------
Please help.

**Comment by howardjohn** [comment]
`Post "https://istiod.istio-system.svc:443/validate?timeout=10s": Forbidden` - Istio will never return Forbidden here AFAIK. so its very likely you have somethign in the networking path returnign this, NOT istio.

I would follow https://github.com/istio/istio/wiki/Troubleshooting-Istio#diagnostics to see where it comes in.

**Comment by yuvrajsinh-chauhan1** [comment]
[root@master ~]kubectl get --raw /api/v1/namespaces/istio-system/services/https:istiod:https-webhook/proxy/inject -v4ect -v4
I0124 10:10:03.103525  495052 helpers.go:246] server response object: [{
  "metadata": {},
  "status": "Failure",
  "message": "error trying to reach service: Forbidden",
  "reason": "ServiceUnavailable",
  "code": 503
}]
Error from server (ServiceUnavailable): error trying to reach service: Forbidden

[root@master ~]# curl https://istiod.istio-system:443/inject -k
curl: (56) Received HTTP code 403 from proxy after CONNECT

[root@master ~]# curl https://localhost:15017/inject -k
curl: (7) Failed to connect to localhost port 15017: Connection refused

Collected Logs -
[istiod.log](https://github.com/istio/istio/files/14033122/istiod.log)
[meshconfig.txt](https://github.com/istio/istio/files/14033169/meshconfig.txt)

**Reply by zirain** [reply]
you may have a network issue between the kube-apiserver to istiod pod.

**Reply by zirain** [reply]
for AWS user, please check firewall and security group.

**Reply by yuvrajsinh-chauhan1** [reply]
I have installed K8s and Istio in Virtual Machine in same network and have not change any network configurations.
If you highlight what to check in K8s network will help.

**Comment by yuvrajsinh-chauhan1** [comment]
Hi All,

Just to update, There was an issue with my VM Proxy settings which is blocking the traffic.
Issue was resolved post allow entry in Proxy settings.

Thanks.
